src/Analyzers/CSharp/Analyzers/UseAutoProperty/CSharpUseAutoPropertyAnalyzer.cs

```diff
@@ -28,83 +39,65 @@ protected override bool SupportsPropertyInitializer(Compilation compilation)
         protected override bool CanExplicitInterfaceImplementationsBeFixed()
             => false;
 
-        protected override void AnalyzeCompilationUnit(
-            SemanticModelAnalysisContext context, SyntaxNode root, List<AnalysisResult> analysisResults)
-            => AnalyzeMembers(context, ((CompilationUnitSyntax)root).Members, analysisResults);
+        protected override ExpressionSyntax? GetFieldInitializer(VariableDeclaratorSyntax variable, CancellationToken cancellationToken)
+            => variable.Initializer?.Value;
 
-        private void AnalyzeMembers(
-            SemanticModelAnalysisContext context,
-            SyntaxList<MemberDeclarationSyntax> members,
-            List<AnalysisResult> analysisResults)
+        protected override void RegisterNonConstructorFieldWrites(
+            ISet<string> fieldNames, ConcurrentDictionary<IFieldSymbol, ConcurrentSet<SyntaxNode>> fieldWrites, SemanticModel semanticModel, SyntaxNode codeBlock, CancellationToken cancellationToken)
         {
-            foreach (var memberDeclaration in members)
-            {
-                AnalyzeMemberDeclaration(context, memberDeclaration, analysisResults);
-            }
+            // nothing to do here for C#.  This is for VB only situations.
         }
 
-        private void AnalyzeMemberDeclaration(
-            SemanticModelAnalysisContext context,
-            MemberDeclarationSyntax member,
-            List<AnalysisResult> analysisResults)
+        protected override void RegisterIneligibleFieldsAction(
+            ISet<string> fieldNames,
```

Comment:
can do a lot less owrk because we can pass in the set of names of things to even bind.